File: graph_theory/floyd_warshall/floyd.go
package main
import "fmt"
func min(a, b int) int {
    if a <= b {
        return a
    }
    return b
}

func main() {
    
    var n,m int
    fmt.Scanf("%v %v\n", &n, &m)
    edges := make([][]int, n+1)

    for i := 0; i <= n ; i++ {
        edges[i] = make([]int, n+1) 
            for j := 0; j <= n; j++ {
                if i == j {
                    edges[i][j] = 0
                } else {
                    edges[i][j] = 999999999999
                }
        }
    }

    var u,v,r int

    for i := 0; i < m  ; i++ {
        fmt.Scanf("%v %v %v\n", &u, &v, &r)
        edges[u][v] = r
    }


    for k := 1; k <= n  ; k++ {
        for i := 1; i <= n  ; i++ {
            for j := 1; j <= n  ; j++ {
                edges[i][j] = min(edges[i][k]+edges[k][j],edges[i][j])
            }
        }
    }

    var q,x,y int
    fmt.Scanf("%v\n", &q)


    for i := 0; i < q; i++ {
        fmt.Scanf("%v %v\n", &x, &y)
        if edges[x][y] == 999999999999 {
            fmt.Println("-1")       
        } else {
            fmt.Println(edges[x][y])
        }

    }
}

File: graph_theory/floyd_warshall/floyd.py
import sys
old_stdin = sys.stdin
sys.stdin = open("/home/maksim/dev_projects/hackerrank/graph_theory/floyd_warshall/test","r")

n,m = list(map(int,input().strip().split()))
edges = [[float("inf") if i !=j else 0 for i in range(n+1)] for j in range(m+1)]
for _ in range(m):
    u,v,r = list(map(int,input().strip().split()))
    edges[u][v] = r
    
for k in range(1,n+1):
    for i in range(1,n+1):
        if edges[i][k] == float("inf"): continue
        for j in range(1,n+1):
            if edges[k][j] == float("inf"): continue
            edges[i][j] = min(edges[i][k]+edges[k][j],edges[i][j])
        
q = int(input().strip())
for _ in range(q):
    x,y = list(map(int,input().strip().split()))
    print(edges[x][y] if edges[x][y] != float("inf") else -1)